pkg/minikube/update/update.go

```diff
@@ -236,3 +288,51 @@ func downloadChecksum(v semver.Version, downloadBinary, downloadLinkFormat strin
 
 	return hex.DecodeString(strings.TrimSpace(string(b)))
 }
+
+func checksumFor(h crypto.Hash, payload []byte) ([]byte, error) {
+	if !h.Available() {
+		return nil, errors.New("requested hash function not available")
+	}
+	hash := h.New()
+	hash.Write(payload) // guaranteed not to error
+	return hash.Sum([]byte{}), nil
+}
+
+func extractAndReplace(goos, dir string, archiveBytes []byte) string {
+	downloadedArchive := filepath.Join(dir, "minishift.tgz")
+	ungzip := filepath.Join(dir, "minishift.gzip")
+	extract := filepath.Join(dir, "minishift-extract")
+	binary := filepath.Join(extract, "minishift")
+
+	// Write archive file to disk
+	err := ioutil.WriteFile(downloadedArchive, archiveBytes, 0644)
```

Comment:
Right. Quite a few glitches there! How about I check the OS first and perform operations based on that? If OS is Windows, save to a file with `.zip` extension else `.tgz` extension? Then extract data out of it and return the path to binary (with `exe` extension on Windows). Modifying the code for your feedback.